.github/ISSUE_TEMPLATE/feature_proposal.md

```diff
@@ -0,0 +1,39 @@
+---
+name: Feature proposal
+about: Propose a great new feature for Mitiq
+title: ''
+labels: 'enhancement'
+assignees: ''
+
+---
+## Pre-Request Checklist
+
+- [ ] I checked to make sure that this feature has not already been requested
+- [ ] I or my organization would be willing to help implement and/or test this.
+- [ ] This is **not** a breaking change to Mitiq.
+
+## Feature description
+
+_TODO: Insert a short outline for what the proposed modification is._
+
+## Considerations
+
+_TODO: Explain why this modification to Mitiq is desirable. Briefly summarize the benefits and drawbacks of the chosen approach compared to other ways of achieving a similar functionality._
+
+## Context
```

Comment:
"Context" also is a bit vague to me. What about (stealing from the description) "Important details" or "Important points going forward"?

Reply:
Also, can we put this section after examples? I think examples are nice to see after the previous section describing the purpose and can elaborate on anything that perhaps wasn't clear there.

Reply:
Reordering is fine with me, and Context really does I think describe what I wanted here (which may not be a helpful thing i get). Basically the question here is are there other things we should consider like other papers or packages that implement part of the feature being proposed? Or is this similar to another technique that we may also want to consider implementing as it would be easy after adding this feature? Have they written up pseudo code for it already and have a link to that? It's the context that this proposal should be considered with. 

Again, maybe we want to know something different from the author and that's totally cool.